### PR TITLE
Bug - 5913 - search page candidate counts

### LINKED ISSUE #5913: 🐛 Search page candidate counts mis-match
## 🐛 Bug

Search page issue again 🙃 

## 🦋 Expected Behaviour

Count per pool should sum to total

## 🕵️ Details

.

## 📋 Steps to Reproduce

1. head to /search, after building off main
2. start adding in filters
3. observe total candidate count and candidate count per pool card

## 📸 Screenshot

image

image

re-seeding

image

## 🙋‍♀️ Proposed Solution

countApplicants is higher in number due to not filtering out suspended candidates

Pool Candidate has a scope for suspended, which is then used in count by pool

image

User does not check for suspended in a scope, this would be done by count applicants

image

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] count per pool sums to total count
- [ ] queries return correct users, as in scopes are correctly assigned

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #5913 

## 👋 Introduction

Resolve the counts not making sense on the search page.

## 🕵️ Details

Needed to incorporate filtering out suspended applications in the count applicant query chain.
Introduced an enum for this, for reuse in issue #5734 

## 🧪 Testing

1. head to /search
2. play around with filters and watch candidate count per pool and total candidate count

🚨 Note 🚨 
users can have multiple pool candidates so total count isn't always the sum of pool counts
but when you've only one pool left, the numbers should match

## 📸 Screenshot

image

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -796,10 +796,17 @@ enum CandidateExpiryFilter {
...
+  ACTIVE
+  SUSPENDED
+  ALL
+}
+
 input UserPoolFilterInput {
   poolId: ID!
   expiryStatus: CandidateExpiryFilter = ACTIVE
   statuses: [PoolCandidateStatus!]
+  suspendedStatus: CandidateSuspendedFilter = ALL
```
Comment (vd1992) [inline api/graphql/schema.graphql:809]: I was a bit iffy on this bit, wasn't sure if it was needed to match this up with the possible fields in `scopePoolFilters`

**Review by yonikid15 — COMMENTED** [review]
Alright nice work Vachan! Took a while to review since the Search Page has so many parts to it. After testing it all out, it looks good to me, the candidate count is showing up as expected. However, regarding `CandidateSuspendedFilter` enum it looks good to me, but it's not part of the AC so I'll let @tristan-orourke approve that.

One issue I found that might be out of scope for this is that after I made a request. The number of estimated users never matched the number of candidates on the "View Request" page admin side. The request table always has more than the requested candidates. I'm not sure exactly why.. yet.

**Comment by vd1992** [comment]
> Note to self - #5929 removes scopeAvailableForOpportunities, how do we ensure that doesn't conflict with this PR?

I don't think it should, the filtering happens in `scopePoolFilters` which is either called directly or by `scopeAvailableInPools`. This PR does not build upon `scopeAvailableForOpportunities`.

File: api/graphql/schema.graphql (hunk 1 above)
Comment (tristan-orourke) [inline api/graphql/schema.graphql:809]: I think this should be set to ACTIVE by default. We won't usually care about suspended candidates.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Hey, great job adding the suspended_at enums! Despite not being specified in the issue, this matches what we've previously done. I'd just change the default filter to ACTIVE.

File: api/graphql/schema.graphql (hunk 1 above)
Comment (vd1992) [reply api/graphql/schema.graphql]: oops, you are right https://github.com/GCTC-NTGC/gc-digital-talent/pull/5918/commits/fefb4a73624a021b24fce5f071c738d754e3107a

File: api/app/Models/User.php [hunk 2]
```diff
@@ -260,6 +261,16 @@ public static function scopePoolFilters(Builder $query, ?array $poolFilters): Bu
                             if (array_key_exists('statuses', $filter) && !empty($filter['statuses'])) {
                                 $query->whereIn('pool_candidates.pool_candidate_status', $filter['statuses']);
                             }
+                            $query->where(function ($query) use ($filter) {
+                                if (array_key_exists('suspendedStatus', $filter) && $filter['suspendedStatus'] == ApiEnums::CANDIDATE_SUSPENDED_FILTER_ACTIVE) {
+                                    $query->where(function ($query) {
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Models/User.php]: think it is https://github.com/GCTC-NTGC/gc-digital-talent/pull/5918/commits/88bc4a68a7b507cf3fb52b5dca6e2742db536369

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**